### ISSUE TITLE
No event emit when watching directory isn't exists before starting

### ISSUE THREAD

**Issue by haishengwu-okta** [paulmillr/chokidar#462]
- Assume `/target` exists and `/target/css` does not exists, then start watching like

```
chokidar.watch(/target/css/**/**.css, {...}, function () {...});
```

I won't get any events when I added either of following
- `/target/css`
- `/target/css/1.css`

---
- Assume `/target/css` exists then everything works fine.

**Comment by ghpabs** [comment]
@haishengwu-okta did you find a workaround for this issue?

**Comment by haishengwu-okta** [comment]
workaround is to create directory in advance, e.g. `target/css`, in an initialization task.
in general you shall know what directory being watched. 
hopefully it's helpful. ;)

**Comment by ghpabs** [comment]
I'm afraid that's not really a solution for me - but that's for the update.

It seems to be somewhat related to https://github.com/paulmillr/chokidar/issues/346.

**Comment by lichenhao** [comment]
The same result with file and not work well with watch file only~

**Comment by paulmillr** [comment]
Update to Chokidar v3, it should resolve this.

**Comment by bdiz** [comment]
Does not seem resolved in 3.5.3.
